plugins/inputs/windows_event_log/wineventlog/utils.go

```diff
@@ -168,3 +170,52 @@ func WindowsEventLogLevelName(levelId int32) string {
 		return UNKNOWN
 	}
 }
+
+// In some cases wevtapi does not insert values when formatting the message. The message
+// will contain insertion string placeholders, of the form %n, where %1 indicates the first
+// insertion string, and so on. Noted that wevtapi start the index with 1.
+// https://learn.microsoft.com/en-us/windows/win32/eventlog/event-identifiers#insertion-strings
+//
+// If we see those data in the `EventData` section, `insertPlaceholderValues` format the message
+// with the correct values
+func insertPlaceholderValues(rawMessage string, evtDataValues []Data) string {
+	if len(evtDataValues) == 0 {
+		return rawMessage
+	}
+	var sb strings.Builder
+	prevIndex := 0
+	searchingIndex := false
+	for i, c := range rawMessage {
+		// found `%` previously. Determine the index number from the following character(s)
+		if searchingIndex && (c > '9' || c < '0') {
```

Comment:
nit: Could use `!unicode.IsDigit(c)`

Reply:
Actually, on second thought, this is fine. I don't think you need to use that function. Simple is better.